app/src/main/java/ch/epfl/sdp/GoogleApi.java

```diff
@@ -0,0 +1,96 @@
+package ch.epfl.sdp;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+
+import androidx.core.app.ActivityCompat;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+public class GoogleApi implements MapApi {

```

Comment:
naming: GoogleApi is too generic, and doesn't provide enough semantics

better examples: `GoogleMapApi` (good practice to keep the suffix of the parent type)